plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1217,24 +1223,51 @@ private void customizePutObjectRequest(PutObjectRequest request)
 
     private InitiateMultipartUploadResult initMultipartUpload(String bucket, String key)
     {
-        InitiateMultipartUploadRequest request = new InitiateMultipartUploadRequest(bucket, key)
-                .withObjectMetadata(new ObjectMetadata())
-                .withCannedACL(s3AclType.getCannedACL())
-                .withRequesterPays(requesterPaysEnabled)
-                .withStorageClass(s3StorageClass.getS3StorageClass());
+        try {
+            return retry()
+                    .maxAttempts(maxAttempts)
+                    .exponentialBackoff(BACKOFF_MIN_SLEEP, maxBackoffTime, maxRetryTime, 2.0)
+                    .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class)
```

Comment:
This should probably stop on `AbortedException.class` as well, which is what the AWS SDK will actually throw if it catches an `InterruptedException` while executing the API request.  Similarly, you might want to reset `Thread.currentThread().interrupt()` for that case too.

Reply:
Follow up https://github.com/trinodb/trino/pull/15578